core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
         return super(FeedbackThreadUserModel, cls).get_multi(instance_ids)
 
 
+class GeneralFeedbackThreadModel(base_models.BaseModel):
+    """Threads for each entity.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # The type of entity the thread is linked to.
+    entity_type = ndb.StringProperty(required=False, indexed=True)
```

Comment:
Why are this and the next field required=False? I would have thought the opposite.

Reply:
Oops. True. Changed.